### ISSUE TITLE
HttpContentEncoder does not handle multiple Accept-Encoding headers correctly

### ISSUE THREAD

**Issue by wilkinsona** [netty/netty#9553]
### Expected behavior

When a request contains multiple `Accept-Encoding` headers, their values will be treated as if they had been sent as the comma-separated values of a single `Accept-Encoding` header.

### Actual behavior

Only the first `Accept-Encoding` header is considered and all subsequent `Accept-Encoding` headers are ignored.

### Steps to reproduce

Configure an HTTP server with `HttpContentCompressor` in its pipeline and send a request with multiple `Accept-Encoding` headers. Using curl that would look something like this:

```
 curl -H "Accept-Encoding: unknown" -H "Accept-Encoding: gzip" localhost:8080
```

Observe that the response is not gzipped.

### Minimal yet complete reproducer code (or URL to code)

```java
package com.example.demo;

import static io.netty.handler.codec.http.HttpHeaderNames.CONNECTION;
import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_LENGTH;
import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE;
import static io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
import static io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
import static io.netty.handler.codec.http.HttpHeaderValues.TEXT_PLAIN;
import static io.netty.handler.codec.http.HttpResponseStatus.OK;

import io.netty.bootstrap.ServerBootstrap;
import io.netty.buffer.Unpooled;
import io.netty.channel.Channel;
import io.netty.channel.ChannelFuture;
import io.netty.channel.ChannelFutureListener;
import io.netty.channel.ChannelHandlerContext;
import io.netty.channel.ChannelInitializer;
import io.netty.channel.ChannelOption;
import io.netty.channel.ChannelPipeline;
import io.netty.channel.EventLoopGroup;
import io.netty.channel.SimpleChannelInboundHandler;
import io.netty.channel.nio.NioEventLoopGroup;
import io.netty.channel.socket.SocketChannel;
import io.netty.channel.socket.nio.NioServerSocketChannel;
import io.netty.handler.codec.http.DefaultFullHttpResponse;
import io.netty.handler.codec.http.FullHttpResponse;
import io.netty.handler.codec.http.HttpContentCompressor;
import io.netty.handler.codec.http.HttpObject;
import io.netty.handler.codec.http.HttpRequest;
import io.netty.handler.codec.http.HttpServerCodec;
import io.netty.handler.codec.http.HttpServerExpectContinueHandler;
import io.netty.handler.codec.http.HttpUtil;

public final class HttpHelloWorldServer {

    public static void main(String[] args) throws Exception {
        EventLoopGroup bossGroup = new NioEventLoopGroup(1);
        EventLoopGroup workerGroup = new NioEventLoopGroup();
        try {
            ServerBootstrap b = new ServerBootstrap();
            b.option(ChannelOption.SO_BACKLOG, 1024);
            b.group(bossGroup, workerGroup)
             .channel(NioServerSocketChannel.class)
             .childHandler(new HttpHelloWorldServerInitializer());
            Channel ch = b.bind(8080).sync().channel();
            ch.closeFuture().sync();
        } finally {
            bossGroup.shutdownGracefully();
            workerGroup.shutdownGracefully();
        }
    }
    
    static final class HttpHelloWorldServerInitializer extends ChannelInitializer<SocketChannel> {

        @Override
        public void initChannel(SocketChannel ch) {
            ChannelPipeline p = ch.pipeline();
            p.addLast(new HttpServerCodec());
            p.addLast(new HttpContentCompressor());
            p.addLast(new HttpServerExpectContinueHandler());
            p.addLast(new HttpHelloWorldServerHandler());
        }
        
    }
    
    static final class HttpHelloWorldServerHandler extends SimpleChannelInboundHandler<HttpObject> {
    	
        private static final byte[] CONTENT = { 'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd' };

        @Override
        public void channelReadComplete(ChannelHandlerContext context) {
            context.flush();
        }

        @Override
        public void channelRead0(ChannelHandlerContext context, HttpObject message) {
            if (message instanceof HttpRequest) {
                HttpRequest req = (HttpRequest) message;

                boolean keepAlive = HttpUtil.isKeepAlive(req);
                FullHttpResponse response = new DefaultFullHttpResponse(req.protocolVersion(), OK, Unpooled.wrappedBuffer(CONTENT));
                response.headers().set(CONTENT_TYPE, TEXT_PLAIN).setInt(CONTENT_LENGTH, response.content().readableBytes());

                if (keepAlive) {
                    if (!req.protocolVersion().isKeepAliveDefault()) {
                        response.headers().set(CONNECTION, KEEP_ALIVE);
                    }
                } else {
                    // Tell the client we're going to close the connection.
                    response.headers().set(CONNECTION, CLOSE);
                }

                ChannelFuture f = context.write(response);

                if (!keepAlive) {
                    f.addListener(ChannelFutureListener.CLOSE);
                }
            }
        }

        @Override
        public void exceptionCaught(ChannelHandlerContext context, Throwable ex) {
            ex.printStackTrace();
            context.close();
        }
        
    }
    
}
```

### Netty version

4.1.39

### JVM version (e.g. `java -version`)

```
openjdk version "1.8.0_202"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_202-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.202-b08, mixed mode)
```

### OS version (e.g. `uname -a`)

Darwin Andys-MacBook-Pro.local 17.7.0 Darwin Kernel Version 17.7.0: Sun Jun  2 20:31:42 PDT 2019; root:xnu-4570.71.46~1/RELEASE_X86_64 x86_64

**Comment by normanmaurer** [comment]
@wilkinsona sounds like a bug... are you interested in providing a PR ?

**Comment by amizurov** [comment]
@normanmaurer  Hi, problem that we handle only first `accept-encoding` header value, also we need to consider the q-value for each, or combine to one header-value by spec https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

**Comment by normanmaurer** [comment]
@amizurov @wilkinsona  I more then happy to review PRs ... Currently I am a bit busy so I will put this on my backlog if no-one else will fix it in the meantime

**Comment by amizurov** [comment]
@normanmaurer No problem if @wilkinsona don't mind I will do a pull request soon.

**Comment by wilkinsona** [comment]
@amizurov I you have the time then please go ahead. If not, I should have some cycles later this week once we've released Spring Boot 2.2 M6.
